docs/filestructure.rst

```diff
@@ -0,0 +1,228 @@
+
+Load Composer autoload.php (which loads up atk4) in init.php
+------------------------------------------------------------
+
+::
+
+  include $rootdir."vendor/autoload.php";   // loads up atk4 and our project files from Composer
+
+Initialize the app class in init.php
+------------------------------------
+
+::
+
+  $app = new \atk4\ui\App('Welcome to my first app'); // initialisation of our app
+  $app->db = $db;   // defines our database for reuse in other classes
+
+Create index.php and admin.php
+------------------------------
+
+If you want to write an app with a backend, create a file called "admin.php"::
+
+  <?php
+  $rootdir = "../";
+  include "init.php";
+  $app->initLayout('Admin');
+
+If you want to write an app with a frontend, create a file called "index.php"::
+
+  <?php
+  $rootdir = "../";
+  include "init.php";
+  $app->initLayout('Centered');
+
+
+Create your own classes
+=======================
+
+Now as your basic app is set up and running, we start implementing our own classes that build the core of our app.
+Following the PSR-4 specifiations all class names and file names have to correspond to each other.
+
+If we want to create a class called "myFirstClass" we have to save it to a file called "myFirstClass.php".
+
+Let's do our first class. Please create a new file in the directory "projectfolder/views" and call it "View1.php".
+
+Now comes a tricky part: you have to define a namespace within your class file that corresponds with the namespace you have
+defined in your composer.json.
+Do you remember? - If no, take a look at the beginning of this document. We defined there "myProject" as our namespace for
+the directory "projectfolder".
+
+Open the created file "View1.php" in your editor and add the following lines::
+
+  <?php
+  namespace myProject\views;
+
+  class View1 extends \atk4\data\View {
+      function init() {
+          parent::init();
+
+          $text = $this->app->add(['Text', 'here goes some text']);
+      }
+  }
+
+"namespace myProject\\views;" defines the namespace to use. It reflects the folder structure of the app.
+The file located in "projectfolder/views/View1.php" becomes "myProject\\views\\View1" in the namespace.
+
+For each of your classes create a separate file. As long as you follow the name conventions all your class
+files will be autoloaded by Composer.
+
+.. warning:: Keep in mind that as soon as you have created one or more new file(s) within the projectfolder you have to run "composer dump-autoload"!!! Otherwise the newly generated file(s) and classes will not be autoloaded and are therefore unavailable in your application.
+
+
+Load your class in index.php
+============================
+
+To use our class in our app, we have to include it into our app. This can be done either through index.php or admin.php.
+
+Please add the following lines into your index.php::
+
+  $app->add(new myProject\views\View1());
```

Comment:
you see if you don't follow naming rules namespaces and classes look strange and not normalized